handler.go

```diff
@@ -34,7 +34,8 @@ func (handler *Handler) ServeHTTP(writer http.ResponseWriter, req *http.Request)
 
 	masterResponseCh := make(chan *Response, 1)
 	responseCh := make(chan *Response, backendCount)
-	done := make(chan bool)
+	allResponsesCh := make(chan []*Response, 1)
```

Comment:
Why do we need a buffered channel? As far as I can tell it can only have a single value at any time.